distribution/src/site/content/conditional-steps.html

```diff
@@ -0,0 +1,114 @@
+                // logic for morning
+                break;
+            case "evening":
+                // logic for evening
+                break;
+            case "afternoon":
+                // logic for evening
+                break;
+        }
+    }
+    </pre>
+
+    <p>With conditional steps the logic can be rewritten in the following way:</p>
+
+    <pre class="brush: java">
+    @Conditional(condition = PartOfDayCondition.class, value = "morning")
+    @Given("food for my pet")
+    public void prepareMorningFood() {
+        // logic for morning
+    }
+
+    @Conditional(condition = PartOfDayCondition.class, value = "evening")
+    @Given("food for my pet")
+    public void prepareEveningFood() {
+        // logic for evening
+    }
+
+    @Conditional(condition = PartOfDayCondition.class, value = "afternoon")
+    @Given("food for my pet")
+    public void prepareAfternoonFood() {
+        // logic for afternoon
+    }
+    </pre>
+
+    <p>To make this work we need the following condition implementation:</p>
+
+    <pre class="brush: java">
+    public class PartOfDayCondition implements Predicate\<Object\>
+    {
+        @Override
+        public boolean test(Object value)
+        {
+            String partOfDay = ...;
+            return partOfDay.equals(value);
+        }
+    }
+    </pre>
+
+    <p>Please note that the condition must have public non-args constructor.</p>
+
+    <p>By default conditions are checked by <a href="javadoc/core/org/jbehave/core/condition/DefaultConditionChecker.html">DefaultConditionChecker</a>,
+        custom conditional checker can be created by implementing the
+        <a href="javadoc/core/org/jbehave/core/condition/ConditionChecker.html">ConditionChecker</a>
+        interface and then registering the new implementation in a
+        <a href="javadoc/core/org/jbehave/core/configuration/Configuration.html">Configuration</a>
+        instance.
+    </p>
+
+    <script type="syntaxhighlighter" class="brush: java">
+    <![CDATA[
+
+    public class MyConditionChecker implements ConditionChecker {
+        @Override
+        public boolean check(Class<? extends Predicate<Object>> type, Object value) {
+            return ...;
+        }
+    }
+
+    public class Stories extends JUnitStories {
+
+        public Stories() {
+            Configuration configuration = new MostUsefulConfiguration()
+                    .useConditionChecker(new MyConditionChecker())
+        }
+    }
+    ]]>
+    </script>
+
+    <p>Notes:</p>
+    <ul>
+        <li>A condition at the method level has higher priority than a condition at the class level</li>
+        <li>if a step is annotated as conditional, then all the steps with the same wording must be conditional,
+            otherwise, an exception is thrown</li>
+        <li>if the condition matches several steps, the first step is used</li>
+        <li>if there are no steps matching, then the step is marked as pending</li>
```

Comment:
I think it's ok to keep it as pending, but the message should clearly state that there were X number of conditional steps, but none of them matched